### DISCUSSION TITLE
ABS could pioneer a in-episode advertisement block for podcasts

### DISCUSSION THREAD

**Discussion by nickelswitte** [advplyr/audiobookshelf#2395] [Ideas]
Hello,
I am a user of [sponsor block](https://sponsor.ajay.app/) on Youtube and really like that I can finally skip many of the annoying in-video advertisements so easily. For anyone who does not know the concept, the first people watching a video will flagg in-video advertisement of the video with its exact timestamps. This information is then stored in a database and all other people will be able to automatically, or with a click of a button, to skip said advertisement by just fast-forwarding to when the ad is over. I can highly recomment to try it out on Youtube to see how it works.

As a regular podcast listener, it becomes obvious to apply this concept also to podcasts and it would not be too much of a technical challenge to implement (in my opinion). I just did a little research and it seems that nothing similar has yet been done for podcasts.

As I am a big fan of ABS, I was thinking if this project could be the first to pioneer this idea? What are your thoughts on this?

**Comment by nichwall** [comment]
This would probably be better suited for a plugin (whenever a plugin system is implemented).

Doing some basic research into how SponsorBlock works, there would need to be a central database to keep track of where the sponsor segments are (there is not a central database or plans to have one for ABS). If people are encoding podcasts at all, which some people do, this affects the timestamps of when advertisements are so the database would also not be a source of truth for everyone. This is already a problem for audiobooks and chapter markers.

ABS is designed as a media server for local files, so if you want to get rid of advertisements from files, you could also look into editing the local files themselves.

This also gets into a bit of whether the functionality would cause unwanted attention for the project (not that this is the same as DRM removal, but similar idea).  https://github.com/advplyr/audiobookshelf/issues/2328#issuecomment-1820888881

**Comment by alex-w-k** [comment]
dynamic ad insertion will make this very difficult especially with DAI being used in the biggest most popular podcasts usually. i don't know how useful that would be. it would work great for podcasts that still use standard in file ads though.
